daemon/src/taker.rs

```diff
@@ -292,3 +242,93 @@ async fn main() -> Result<()> {
 
     Ok(())
 }
+
+pub struct ActorSystem<O, M> {
+    cfd_actor_addr: Address<taker_cfd::Actor<O, M>>,
+    cfd_feed_receiver: watch::Receiver<Vec<Cfd>>,
+    order_feed_receiver: watch::Receiver<Option<Order>>,
+    update_cfd_feed_receiver: watch::Receiver<UpdateCfdProposals>,
+}
+
+impl<O, M> ActorSystem<O, M>
+where
+    O: xtra::Handler<oracle::MonitorAttestation>
+        + xtra::Handler<oracle::GetAnnouncement>
+        + xtra::Handler<oracle::FetchAnnouncement>
+        + xtra::Handler<oracle::Sync>,
+    M: xtra::Handler<monitor::StartMonitoring>
+        + xtra::Handler<monitor::Sync>
+        + xtra::Handler<monitor::CollaborativeSettlement>,
+{
+    pub async fn new<F>(
+        db: SqlitePool,
+        wallet: Wallet,
+        oracle_pk: schnorrsig::PublicKey,
+        send_to_maker: Box<dyn MessageChannel<wire::TakerToMaker>>,
+        read_from_maker: Box<dyn Stream<Item = taker_cfd::MakerStreamMessage> + Unpin + Send>,
+        oracle_constructor: impl Fn(Vec<Cfd>, Box<dyn StrongMessageChannel<Attestation>>) -> O,
+        monitor_constructor: impl Fn(Box<dyn StrongMessageChannel<monitor::Event>>, Vec<Cfd>) -> F,
+    ) -> Self
+    where
+        F: Future<Output = Result<M>>,
+    {
+        let mut conn = db.acquire().await.unwrap();
+
+        let cfds = load_all_cfds(&mut conn).await.unwrap();
+
+        let (cfd_feed_sender, cfd_feed_receiver) = watch::channel(cfds.clone());
```

Comment:
Is it worth it to do this inside here if we are passing the `cfd_feed_receiver` out again? We could just do this outside and only pass the `sender` in?